class/defaults.yml

```diff
@@ -11,6 +11,30 @@ parameters:
         repository: kyverno/kyvernopre
         version: ${kyverno:images:kyverno:version}
 
+    manifest_version: ${kyverno:images:kyverno:version}
+    manifest_url: https://raw.githubusercontent.com/kyverno/kyverno/${kyverno:manifest_version}/definitions
+
+    replicas: 3
+
+    podDisruptionBudget:
+      maxUnavailable: 1
```

Comment:
"Thinking" about PDB != "Making" one :)

I think `minAvailable=1` would be the better parameter. Since we don't need Quorum and HA is implemented Active-Passive with Leader Election, we only need 1 running pod.

Reply:
Also, does it make sense to create a parameter `enabled=true`. This would allow (temporarily) removing the PDB if there is a reason to (e.g. maintenance or incidents)

Reply:
> we don't need Quorum

fair point

Reply:
This maybe needs testing though. I have the sneaking suspicion that the PDB will prevent draining the node where the Kyverno leader pod is running, because the standby pods my not be ready (failing readiness probe, since they aren't leader. If they would be leaders, they would get traffic).
And to make them ready, the leader pod needs to give up the leader.

This is a assumption though, I don't know if that's the case.

I could not find any documentation or blogs that could answer my question whether PDBs consider the liveness or readiness state, or only if the pod is in state "Running". Do you know more?

Reply:
Master identity is coordinated by leases. All kyverno replicas are ready:
```
NAME                       READY   STATUS    RESTARTS   AGE
kyverno-7dc8969bfc-gzl7p   1/1     Running   6          4d
kyverno-7dc8969bfc-hzq4k   1/1     Running   0          4m16s
kyverno-7dc8969bfc-z8frc   1/1     Running   0          4m17s
``` 

The readiness state is considered:

```
❯ k get pods
NAME                                      READY   STATUS    RESTARTS   AGE
readiness-test-ready-7b9bfb6546-xzwq9     1/1     Running   0          2m15s
readiness-test-unready-77f99b8879-lcgcl   0/1     Running   0          2m15s

❯ k get pdb readiness-test -o jsonpath='{.status.currentHealthy}' 
1
```

Reply:
Ah, it's good that the Kyverno pods are ready. So we can use the PDB to keep at least 1 running.